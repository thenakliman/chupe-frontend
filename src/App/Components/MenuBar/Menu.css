* {
  padding: 0;
  margin: 0;
}

.Chupe {
}

.Header {
    position: fixed;
    background-color: #20232a;
    color: #fff;
    width: 100%;
    padding: .5rem;
    text-align: left;
    font-weight: 700;
    font-size: 1.5rem;
}
