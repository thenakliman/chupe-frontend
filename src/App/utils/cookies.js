import Cookies from 'universal-cookie';
import {verify, decode} from 'jsonwebtoken';

const TOKEN_COOKIE_KEY = 'token';
const TOKEN_EXPIRY_COOKIE_KEY = 'expiryTime';
const TOKEN_USERNAME_KEY = 'username';
const audience = 'chupe-frontend';
const issuer = 'chupe';
export let cookies = new Cookies();
const Config = require('Config');

const setCookie = (key, value) => {
  cookies.set(key, value);
};

export const setCookies = (token) => {
  verify(token,
         Config.signingKey,
         {
            algorithms: ['HS512'],
            audience: audience,
            issuer: issuer,
            ignoreExpiration: false,
         },
         (e) => console.log(e)
  );

  const decodedToken = decode(token);
  setToken(token);
  setTokenExpiryTime(decodedToken.exp);
  setUsername(decodedToken.username);
};

const setToken = (token) => {
  setCookie(TOKEN_COOKIE_KEY, token);
};

const setTokenExpiryTime = (expiryTime) => {
  setCookie(TOKEN_EXPIRY_COOKIE_KEY, expiryTime);
};

const setUsername = (userName) => {
  setCookie(TOKEN_USERNAME_KEY, userName);
};

export const getUsername = () => {
  return cookies.get(TOKEN_USERNAME_KEY);
};

export const getToken = () => {
  if (!cookies.get(TOKEN_COOKIE_KEY) ||
      !cookies.get(TOKEN_EXPIRY_COOKIE_KEY) ||
      cookies.get(TOKEN_EXPIRY_COOKIE_KEY) < Date.now()) {
    return null;
  }

  return cookies.get(TOKEN_COOKIE_KEY);
};
