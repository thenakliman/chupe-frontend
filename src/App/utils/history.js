import {createHashHistory} from 'history';

export let history = createHashHistory();
