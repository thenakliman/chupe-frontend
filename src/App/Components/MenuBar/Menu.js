import React from 'react';
import {removeCookies} from '../../utils/cookies';
import {history} from '../../utils/history';
import PropTypes from 'prop-types';
import {LOCATION_TO_TAB_MAPPING} from './constants';

require('./Menu.css');


/**
* User result component of the application.
*
* @author: thenakliman
*/
export class MenuBar extends React.Component {
    /** logout user session. */
    logout() {
      removeCookies();
      history.push('/');
    }

    /**
     * Checks tab for a given location.
     * @param {string} location to check for
     * @param {string} tabName to check if tab belongs to location
     * @return {bool} returns whether a tab belong to given location or not
     */
    doesLocationMatchTab(location, tabName) {
      const locationPrefix = location.split('/')[1];
      return LOCATION_TO_TAB_MAPPING[tabName].some(
          (tab) => tab.startsWith(locationPrefix));
    }
    /**
    * User result component of the application.
    *
    * @return {Object} UserResult component.
    */
    render() {
      const location = this.props.location.pathname;
      return (
        <div>
          <header id="chupe-header" className="Header">
            <span className='manu-bar-tabs'>
              <a href='#/dashboard'
                 id="home-tab"
                 className={this.doesLocationMatchTab(location, 'home')?
                            'Selected-Tab':
                            'Tab'}>
                     Home
              </a>
              <a href='#/users'
                 id="users-tab"
                 className={this.doesLocationMatchTab(location, 'users')?
                            'Selected-Tab':
                            'Tab'}>
                     Users
              </a>
              <a href='#/questions'
                 id="questions-tab"
                 className={this.doesLocationMatchTab(location, 'questions')?
                            'Selected-Tab':
                            'Tab'}>
                   Questions
              </a>
              <a href='#/team-funds'
                 id="team-funds-tab"
                 className={this.doesLocationMatchTab(location, 'team-funds')?
                            'Selected-Tab':
                            'Tab'}>
                   Funds
              </a>
              <a href='#/tasks'
                 id="tasks-tab"
                 className={this.doesLocationMatchTab(location, 'tasks')?
                            'Selected-Tab':
                            'Tab'}>
                   Tasks
              </a>
              <a href='#/retros'
                 id="retros-tab"
                 className={this.doesLocationMatchTab(location, 'retros')?
                            'Selected-Tab':
                            'Tab'}>
                   Retrospections
              </a>
              <a href='#/feedback-sessions'
                 id="feedback-session-tab"
                 className={this.doesLocationMatchTab(
                    location, 'feedbackSessions')?
                            'Selected-Tab':
                            'Tab'}>
                   Feedbacks
              </a>
              <a href='#/meetings'
                 id="meetings-tab"
                 className={this.doesLocationMatchTab(
                    location, 'meetings')?
                            'Selected-Tab':
                            'Tab'}>
                   Meetings
              </a>
            </span>
            <span className='manu-bar-tabs'>
              <div id='menu-logout-container'
                   className='menu-logout-container'
                   onClick={this.logout}>
                logout
              </div>
            </span>
          </header>
        </div>
        );
    }
}


MenuBar.propTypes = {
  location: PropTypes.object.isRequired,
};
