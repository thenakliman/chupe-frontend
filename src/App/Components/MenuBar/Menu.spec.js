import React from 'react'; // eslint-disable-line no-unused-vars
import {MenuBar} from './Menu'; // eslint-disable-line no-unused-vars
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import * as cookies from '../../utils/cookies';
import * as History from '../../utils/history';

describe('Menu Bar Snapshot', () => {
  it('should match snapshot', () => {
    const location = {pathname: '/users'};
    const wrapper = shallow(
        <MenuBar
            showQuestionTab={()=>{}}
            showUserTab={()=>{}}
            location={location}
        />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});

describe('Menu Bar Component', () => {
  const location = {pathname: '/users'};

  it('Should display div with header class', () => {
    const wrapper = shallow(
        <MenuBar
            showUserTab={()=>{}}
            location={location}
            showQuestionTab={()=>{}}
        />);
    expect(wrapper.find('.Header').get(0).props.id).toEqual('chupe-header');
  });

  it('Should display div logout', () => {
    const wrapper = shallow(
        <MenuBar
            showUserTab={()=>{}}
            location={location}
            showQuestionTab={()=>{}}
        />);
    expect(wrapper.find('#menu-logout-container').length).toEqual(1);
  });

  it('Should should call removeCookies on click of logout', () => {
    spyOn(cookies, 'removeCookies');
    const pushMock = jest.fn();
    History.history = {push: pushMock};
    const wrapper = shallow(
        <MenuBar
            showUserTab={()=>{}}
            location={location}
            showQuestionTab={()=>{}}
        />);

    wrapper.find('#menu-logout-container').simulate('click');

    expect(cookies.removeCookies).toHaveBeenCalledWith();
    expect(pushMock).toHaveBeenCalledWith('/');
  });

  it('Should have user class by default', () => {
    const location = {pathname: '/users'};
    const wrapper = shallow(
        <MenuBar
            showUserTab={()=>{}}
            showQuestionTab={()=>{}}
            location={location}
        />);
    expect(wrapper.find('#users-tab').props().className
      ).toEqual('Selected-Tab');
  });

  it('Should have Team Fund tab', () => {
    const location = {pathname: '/team-funds'};
    const wrapper = shallow(
        <MenuBar
            showUserTab={()=>{}}
            showQuestionTab={()=>{}}
            location={location}
        />);
    expect(wrapper.find('#team-funds-tab').length).toEqual(1);
  });
});
