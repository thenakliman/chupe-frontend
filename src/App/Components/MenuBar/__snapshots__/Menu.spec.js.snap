// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu Bar Snapshot should match snapshot 1`] = `
<div>
  <header
    className="Header"
    id="chupe-header"
  >
    <span
      className="manu-bar-tabs"
    >
      <a
        className="Selected-Tab"
        href="#/users"
        id="users-tab"
      >
        Users
      </a>
      <a
        className="Tab"
        href="#/questions"
        id="questions-tab"
      >
        Questions
      </a>
      <a
        className="Tab"
        href="#/team-funds"
        id="team-funds-tab"
      >
        Team Funds
      </a>
    </span>
    <span
      className="manu-bar-tabs"
    >
      <div
        className="menu-logout-container"
        id="menu-logout-container"
        onClick={[Function]}
      >
        logout
      </div>
    </span>
  </header>
</div>
`;
